systemdev/kernel-hal.rst

```diff
@@ -0,0 +1,64 @@
+Kernel and hardware abstraction
+===============================
+
+This page documents the resources and process to build the kernel and hardware abstraction. This document is useful if you would like to work on:
+
+* A hardware related issue (camera, sensors, radios)
+* A Linux kernel related issue
+* An issue related to the system-image upgrade process
+
+This document is not useful if you would like to modify :doc:`../contribute/preinstalled-apps` or :doc:`System software <../systemdev/index>`. See the respective documentation for each.
+
+
+
+We distinguish a few different groups of Ubuntu Touch devices with respect to how the kernel and hardware abstraction is implemented:
+
+  * Android 5.1 based ports
+  * Halium based ports
+  * Linux kernel
+
+Android 5.1 based ports
+-----------------------
+
+Android 5.1 based ports of Ubuntu Touch consist of the Linux kernel for this device plus a minimal Android system that is used to enable all the hardware. Every device has it's own fork of the Linux kernel and they all quite old and heavily modified for the specific device by the manufacturer of that device.
```

Comment:
Consider: "Every device has it's own fork on the Linux kernel and they are all heavily modified for the specific hardware specification by the manufacturer of that device"

Reply:
pls recheck